### ISSUE TITLE
Utilize Pydantic for configuration validation

### ISSUE THREAD

**Issue by stranger-danger-zamu** [DIGITALCRIMINALS/UltimaScraper#226]
In the pursuit of better readability, I went through and wrote the entire config parsing code (`./classes/make_settings.py`) using [pydantic](https://github.com/samuelcolvin/pydantic). It's not only shorter by half, but it also is more strict. The strictness however is a boon as it *actually* tells you where your settings are wrong. 

```python
from pydantic import BaseModel, Field
from enum import Enum
from pydantic.types import DirectoryPath, FilePath
from pathlib import Path
from typing import Optional
import uuid


class StrEnum(str, Enum):
    pass

class ExportType(StrEnum):
    json = 'json'

class MediaChoiceType(StrEnum):
    ...

class VideoQualityType(StrEnum):
    source = 'source'

class WebhookConfig(BaseModel):
    webhooks: list[str]
    status: Optional[str]
    hide_sensitive_info: bool = True

class AuthWebhookConfig(BaseModel):
    succeeded: WebhookConfig
    failed: WebhookConfig

class DownloadWebhookConfig(BaseModel):
    succeeded: WebhookConfig

class WebhookSetting(BaseModel):
    global_webhooks: list[str]
    global_status: bool = True
    auth_webhook: AuthWebhookConfig
    download_webhook: DownloadWebhookConfig

class HelperSetting(BaseModel):
    renamer: bool = True
    reformat_media: bool = True
    downloader: bool = True
    delete_empty_directories: bool = False

class GeneralSettingSection(BaseModel):
    auto_site_choice: str = ""
    export_type: ExportType
    profile_directories: list[Path]
    max_threads: int
    min_drive_space: float
    helpers: HelperSetting
    webhooks: WebhookSetting
    exit_on_completion: bool = False
    infinite_loop: bool = True
    loop_timeout: int = 0
    dynamic_rules_link: str = "https://raw.githubusercontent.com/DATAHOARDERS/dynamic-rules/main/onlyfans.json"
    proxies: list[str]
    cert: Optional[Path] = None
    random_string: str = Field(default_factory=lambda: uuid.uuid1().hex)

class InfoSection(BaseModel):
    version: float

class BrowserSettingSection(BaseModel):
    auth: bool = True

class JobsSettingSection(BaseModel):
    scrape_names: bool = True
    scrape_paid_content: bool = True

class SupportedSite(BaseModel):
    auto_profile_choice: list[str]
    auto_model_choice: list[str]
    auto_media_choice: list[str]
    auto_api_choice: bool = True
    browser: BrowserSettingSection
    jobs: JobsSettingSection
    download_directories: list[Path]
    file_directory_format: str = "{site_name}/{model_username}/{api_type}/{value}/{media_type}"
    filename_format: str = "{filename}.{ext}"
    metadata_directories: list[Path]
    metadata_directory_format: str = "{site_name}/{model_username}/Metadata"
    delete_legacy_metadata: bool = False
    text_length: int = 255
    video_quality: VideoQualityType = 'source'
    overwrite_files: bool = False
    date_format: str = '%d-%m-%Y'
    ignored_keywords: list[str]
    ignore_type: list[str]
    blacklists: list[str]
    webhook: bool = True

class Site(BaseModel):
    settings: SupportedSite

class SupportedSitesSection(BaseModel):
    onlyfans: Site
    fansly: Site
    starsavn: Site

class Settings(BaseModel):
    info: InfoSection
    settings: GeneralSettingSection
    supported: SupportedSitesSection

if __name__=="__main__":
    from pydantic.error_wrappers import ValidationError
    with open('./.settings/config.json') as f:
        data = f.read()
    import json
    try:
        Conf = Settings.parse_obj(json.loads(data))
        print(Conf.json())
    except ValidationError as e:
        print(e)
```
The only additional direct dependency to test this out is pydantic. You will probably need to change some values in your config.json file to match the type requirements (mainly supported sites now enforcing lists instead of accepting `bool`s, `list`s, and `str`s for some attributes).

There is some useless things like Supported Sites having only one attribute which houses the settings, but I left that in since that'd probably break a lot of code to change. 

Also I did include some enums but not all values are filled out for them since I couldn't find a list of valid values (ie. the README.md only provides some not all valid values). 

---

tl;dr: rewrote validation using pydantic. above code is standalone, but works with minor changes to config.json. should be simple to incorporate it, but it will break stuff for end users (but when doesn't this project lol).

Also it shouldn't be too hard to keep the API exposed by make_settings.py and utilize this, but I didn't have the time for that right now.
